### ISSUE TITLE
dropout layer inside fpn?

### ISSUE THREAD

**Issue by im9uri** [matterport/Mask_RCNN#72]
Inside the method of fpn_classifier_graph, there is a dropout layer between the two fc layers.
I read the paper and couldn't find if the original implementation uses dropout.

My current understanding is that the dropout layer is there because the parameter count is really large, is this correct?

**Comment by jiang1st** [comment]
And BN is not mentioned in the original paper, neither.

**Comment by waleedka** [comment]
@im9uri You're right. While building the model, among many other experiments, I added the dropout layer to see if it improved results. I just pushed a few updates that remove the dropout and improve other things. I also uploaded [a new trained model](https://github.com/matterport/Mask_RCNN/releases/tag/v2.0) that gets higher accuracy. 

@jiang1st I'll need to check that. Thanks! Maybe I just added the BN layers out of habit. I don't think they cause any hard, though, because all BN layers are frozen anyway.

**Comment by jiang1st** [comment]
@waleedka How did you freeze all BN layers? in layer_regex, bn layers are included.

**Comment by waleedka** [comment]
Keras treats BN layers differently. If you set `layer.training = False` it only freezes the beta and gamma weights, but it doesn't freeze the running mean and variance. I think that's not the best approach, but it is what it is. The only way to freeze all weights of a BN layer in Keras is by passing `training=False` when you construct the layer, and it cannot be changed later by setting the `layer.training` flag. For these reasons, I disable training on all BN layers by sub-classing it and hard-wiring it to be frozen. This is the relevant part of the code from `model.py`

```
class BatchNorm(KL.BatchNormalization):
    """Batch Normalization class. Subclasses the Keras BN class and
    hardcodes training=False so the BN layer doesn't update
    during training.

    Batch normalization has a negative effect on training if batches are small
    so we disable it here.
    """
    def call(self, inputs, training=None):
        return super(self.__class__, self).call(inputs, training=False)
```
